extensions/src/org/pentaho/platform/osgi/KarafInstance.java

```diff
@@ -139,7 +136,7 @@ private void assignInstanceNumber() {
     }
 
     instanceNumber = testInstance;
-
+    System.setProperty( "karaf.data", cachePath );
```

Comment:
Why are we inconditionally setting the system property. If it was set from outside, we shouldn't override it, right ?